#409 Implement timeout support to networking Client

mnylen opened the pull request:
### Changes

- Added timeout option to `Auth0` that is passed to `networking.Client` (defaults to 10 seconds)
- Requests made with `networking.Client` timeout after the specified timeout, and any ongoing requests are cancelled using `AbortController` created for each request.
- Exports `TimeoutError`

### References

Please include relevant links supporting this change such as a:

- https://github.com/auth0/react-native-auth0/issues/409

### Testing

Can be tested in e.g. iOS Simulator using Network Link Condition preferences pane. Setting the packet drop ratio to 100 % will cause the requests to never complete. With this change, `TimeoutError` will be thrown after attempting the request for 10 seconds.

- [x] This change adds unit test coverage
- [ ] This change has been tested on the latest version of the platform/language or why not.

### Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [x] All existing and new tests complete without errors
- [x] All active GitHub checks have passed

Widcket commented:
Hi @mnylen, thanks for raising this PR. I haven't been able to look at it yet, but might get to it by the end of the week.

Widcket commented on src/networking/__tests__/index.spec.js:
```diff
@@ -233,8 +234,53 @@ describe('client', () => {
+      baseUrl,
+      telemetry: {name: 'react-native-auth0', version: '1.0.0'},
+      token: 'a.bearer.token',
+      timeout: 10,
```
Please use a lower value.

Widcket reviewed (commented):
Looks good, just have a nitpick over the timeout value for the test case. Thank you for the PR, we really appreciate you taking the time for doing this.

mnylen commented:
@Widcket thanks for the feedback. I changed the test timeout. Sorry it took a while. Switched companies and not in a project that uses this anymore and don't have the setup locally on my computer anymore either, but hopefully others can find this change useful. :)

Widcket commented:
Thanks for coming back and pushing the change!
